tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,414 @@
+                create_attributes(
+                    dataset=dataset, attributes=[attribute_name], attribute_types=attribute_types
+                )
+
+        # Remove any attributes from dataset that aren't in the new list of attributes
+        for attribute_name in existing_attributes:
+            if attribute_name not in attributes and attribute_name not in primary_keys:
+                delete_attributes(dataset=dataset, attributes=[attribute_name])
+
+    return dataset
+
+
+def create_attributes(
+    *,
+    dataset: Dataset,
+    attributes: List[str],
+    attribute_types: Optional[Dict[str, attribute_type.AttributeType]] = None,
+) -> Dataset:
+    """Creates attributes in dataset if they don't already exist.
+       If no attribute_types are passed in, the default will be ARRAY STRING
+
+    Args:
+        dataset: An existing TUC dataset
+        attributes: list of attribute names to add to dataset
+        attribute_types: dict of attribute types, attribute name is key and AttributeType is value
+
+    Returns:
+        Updated Dataset
+
+    Raises:
+        requests.HTTPError: If any HTTP error is encountered
+        TypeError: attributes arg must be a List
+        ValueError: trying to alter a unified dataset
+        ValueError: An attribute with name '{attribute_name}' already exists in {dataset_name}
+    """
+    dataset_name = dataset.name
+    if dataset.upstream_datasets():
+        raise ValueError(f"{dataset_name} is not a source dataset")
+
+    # Check input type is correct
+    if type(attributes) != list:
+        raise TypeError("attributes arg must be a List")
+
+    # Get current dataset attributes
+    target_dataset_attributes = dataset.attributes
+    existing_attributes = []
+    for attr in target_dataset_attributes.stream():
+        existing_attributes.append(attr.name)
+
+    # Check that none of the new attribute names already exist
+    for attribute_name in attributes:
+        if attribute_name in existing_attributes:
+            # This attribute already exists
+            raise ValueError(
+                f"An attribute with name '{attribute_name}' already exists in {dataset_name}"
+            )
+
+    # Add attributes to dataset
+    for attribute_name in attributes:
+        attr_spec_dict = _make_spec_dict(
+            attribute_name=attribute_name, attribute_types=attribute_types
+        )
+        target_dataset_attributes.create(attr_spec_dict)
+        LOGGER.info(f"Created attribute '{attribute_name}' in {dataset_name}")
+
+    return dataset
+
+
+def edit_attributes(
+    *,
+    dataset: Dataset,
+    attributes: List[str],
+    attribute_types: Optional[Dict[str, attribute_type.AttributeType]] = None,
+    attribute_descriptions: Optional[JsonDict] = None,
+    override_existing_types: bool = False,
+) -> Dataset:
+    """Edits existing attributes in dataset.
+       If an attrbute_type is not defined the default will be ARRAY STRING
```

Comment:
typo
```suggestion
       If an attribute_type is not defined the default will be ARRAY STRING
```

Reply:
I also don't think we should restrict this to being a list. What if someone passes a tuple, set, or other iterable?  Instead let's use `isinstance(t, Iterable)`, with `from typing import Iterable` (and updating the type-hint).